src/Stratis.Bitcoin.IntegrationTests/BlockStore/ReOrgRegularlySpecification_Steps.cs

```diff
@@ -0,0 +1,159 @@
+
+            this.nodeBuilder.StartAll();
+
+            this.selfishMiner.NotInIBD();
+            this.secondNode.NotInIBD();
+            this.thirdNode.NotInIBD();
+            this.fourthNode.NotInIBD();
+
+            this.selfishMiner.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.secondNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.thirdNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.fourthNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+
+            this.selfishMiner.CreateRPCClient().AddNode(this.secondNode.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.secondNode, this.selfishMiner));
+
+            this.secondNode.CreateRPCClient().AddNode(this.thirdNode.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.thirdNode, this.secondNode));
+
+            this.thirdNode.CreateRPCClient().AddNode(this.fourthNode.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.fourthNode, this.thirdNode));
+        }
+
+        private void each_mine_a_blocks()
+        {
+            this.sharedSteps.MineBlocks(1, this.selfishMiner, AccountZero, WalletZero, WalletPassword);
+            this.sharedSteps.MineBlocks(1, this.secondNode, AccountZero, WalletZero, WalletPassword);
+            this.sharedSteps.MineBlocks(1, this.thirdNode, AccountZero, WalletZero, WalletPassword);
+            this.sharedSteps.MineBlocks(1, this.fourthNode, AccountZero, WalletZero, WalletPassword);
+        }
+
+        private void selfish_miner_disconnects_and_mines_10_blocks()
+        {
+            this.selfishMiner.FullNode.ConnectionManager.RemoveNodeAddress(this.secondNode.Endpoint);
+            this.selfishMiner.FullNode.ConnectionManager.RemoveNodeAddress(this.thirdNode.Endpoint);
+            this.selfishMiner.FullNode.ConnectionManager.RemoveNodeAddress(this.fourthNode.Endpoint);
+            TestHelper.WaitLoop(() => !TestHelper.IsNodeConnected(this.selfishMiner));
+
+            this.sharedSteps.MineBlocks(10, this.selfishMiner, AccountZero, WalletZero, WalletPassword);
+
+            this.selfishBlockHeight = this.selfishMiner.FullNode.Chain.Height;
+        }
+
+        private void second_node_creates_a_transaction_and_broadcasts()
+        {
+            var thirdNodeReceivingAddress = this.thirdNode.FullNode.WalletManager()
+                .GetUnusedAddress(new WalletAccountReference(WalletZero, AccountZero));
+
+            this.secondNodeTransaction = this.secondNode.FullNode.WalletTransactionHandler().BuildTransaction(
+                SharedSteps.CreateTransactionBuildContext(new WalletAccountReference(WalletZero, AccountZero), WalletPassword, thirdNodeReceivingAddress.ScriptPubKey,
+                    Money.COIN * 1, FeeType.Medium, 101));
```

Comment:
What does 101 mean ?

Reply:
not sure it is relevant to this test but presumably the number of confirmations needed for the wallet to allow a spend? Perhaps because I took this from a wallet test....  and 101 implies something to do with maturity. Prob because the test is spending a mined coin

Reply:
yes I think thats it, probably we need to fix the wallet to not allow spending mined coins, but I am not sure the wallet has that info (it will have to flag such an output)